test_server.py

```diff
@@ -0,0 +1,20 @@
+import unittest
+import pytest
+#import server as app
+import requests
+
+url = 'http://localhost:8080'
+
+def test_question():
+    r = requests.get(url+'/question/1')
+    assert r.status_code == 200
+
+def test_download():
+    assert requests.get(url+'/question/1/inputs.txt').status_code == 200
+
+def test_file_upload():
+    files= open('files/questions/1/output.txt')
```

Comment:
Why are you doing this?

Reply:
Actually tell me how you decided to write this line. Maybe I can help you then.

Reply:
i have no idea what i thought then, i just wanted to create a test function which can verify that file is posted.

Reply:
let me rephrase.

What steps did you follow to write that test function. Here's what I usually do:

1. google for how to upload file
2. find documentation and read it
3. try to find an exact example of what I'm trying to do
4. understand the parts of the example that are relevant
5. copy paste into my code

Now. What did you do?

Reply:
for example, if you followed a similar list, how did you find the example of what you are trying to do?

Reply:
I did the exact above things, I was not able to find out the exact example.
but as you say I will do it once again and will discuss the example with you.
sounds cool?

Reply:
I think I linked to the exact place in the document which we needed to implemented?
![image](https://user-images.githubusercontent.com/7693265/55701274-2fc56700-59f0-11e9-9c4c-3e49f7ce6425.png)

Reply:
```
def test_file_upload():
    files= {'file' : open('files/questions/1/output.txt', 'rb')}
    global url
    url += '/check/1/output.txt'

    assert requests.post(url, files = files).status_code == 200   # givin a error
```

is this correct?

Reply:
is it running?

Reply:
no, this is what i am getting.
```
platform linux -- Python 3.7.2, pytest-3.8.0, py-1.6.0, pluggy-0.7.1
rootdir: /home/shivank98/Desktop/PyJudge/PyJudge, inifile:
plugins: remotedata-0.3.0, openfiles-0.3.0, doctestplus-0.1.3, arraydiff-0.2, celery-4.2.1
collected 3 items

test_server.py ..F                                  [100%]

======================== FAILURES =========================
____________________ test_file_upload _____________________

    def test_file_upload():
        files= {'file' : open('files/questions/1/output.txt', 'rb')}
        global url
        url += '/check/1/output.txt'

>       assert requests.post(url, files = files).status_code == 200   # givin a error
E       AssertionError: assert 404 == 200
E        +  where 404 = <Response [404]>.status_code
E        +    where <Response [404]> = <function post at 0x7fd95ef0b268>('http://localhost:8080/check/1/output.txt', files={'file': <_io.BufferedReader name='files/questions/1/output.txt'>})
E        +      where <function post at 0x7fd95ef0b268> = requests.post

test_server.py:20: AssertionError
```

Reply:
please debug. The error message is on your screen. I can see the mistake but it might be better if you find it on your own.

Reply:
please just tell me, so I can just end this issue and start working from fresh on a new issue. i know i can do it my own too. but ...

Reply:
No.

The point is for you to learn. You have to go to the gym and lift your own weights. If someone else does it for you, going to the gym is useless.

Take your time. There is no time limit. Do it right.

Reply:
Ohk,

Reply:
[Dashrath Manjhi](https://en.wikipedia.org/wiki/Dashrath_Manjhi) ko yaad rakho. Pahaad toda tha unhone! 20 saal lage the. Is issue ko toh sirf ek hafta hua hai :smile: Lage raho janaab, jo gyaan ka phool uge ga na aap mein aage ki zindagi aasaan ho jaayegi

Reply:
give me some hints so I can do it asap.

Reply:
@theSage21 i can see the given url is incorrect.
for `url += '/question/1'` i am receiving the `405`

Reply:
@Shivank98 Shouldn't the check url be `/check/1` instead of `/check/1/output.txt`
According to me the program is giving 404 for not finding the handler for `/check/1/output.txt`

Reply:
code
```
def test_file_upload():
    file= {'file' : open('files/questions/1/output.txt', 'rb')}
    
    global url
    url += '/check/1'
    response = requests.post(url, data=file)
    assert response.status_code == 200
```
error
```
def test_file_upload():
        file= {'file' : open('files/questions/1/output.txt', 'rb')}

        global url
        url += '/check/1'
        response = requests.post(url, data=file)
>       assert response.status_code == 200
E       assert 413 == 200
E        +  where 413 = <Response [413]>.status_code

test_server.py:26: AssertionError
```

Reply:
would you like to pick up some other issue? Maybe someone else can do this?

Reply:
Got it!
Use,
```
def test_file_upload():
    files= {'upload' : open('files/questions/1/output.txt', 'rb')}
    global url
    url += '/check/1'
    assert requests.post(url, files = files).status_code == 200

```

Reply:
damn it, is it what I was missing.?
fool me.